StreamChatClient_v3/Controllers/ChannelController.swift

```diff
@@ -5,78 +5,300 @@
+    /// The ChannelQuery this controller observes.
+    public let channelQuery: ChannelQuery<ExtraData>
+    
+    /// The `ChatClient` instance this controller belongs to.
+    public let client: Client<ExtraData>
     
-    /// Acts like the `NSFetchedResultsController`'s delegate and aggregates the reported changes into easily consumable form.
-    private(set) lazy var changeAggregator: ChangeAggregator<ChannelDTO, ChannelModel<ExtraData>> = {
-        let aggregator: ChangeAggregator<ChannelDTO, ChannelModel<ExtraData>> = .init(itemCreator: ChannelModel<ExtraData>.create)
+    /// The channel matching the channelId. To observe updates to the channel, set your class as a delegate of this controller and call `startUpdating`.
+    public var channel: ChannelModel<ExtraData>? {
+        guard state == .active else {
+            log.warning("Accessing `channel` before calling `startUpdating()` always results in `nil`.")
+            return nil
+        }
         
-        aggregator.onChange = { [unowned self] (_: [Change<ChannelModel<ExtraData>>]) in
-            guard let channel = self.fetchResultsController.fetchedObjects?.first
-                .map(ChannelModel<ExtraData>.create(fromDTO:)) else { return }
-            
-            self.delegate?.channelController(self, didUpdateChannel: channel)
+        return channelObserver.item
+    }
+    
+    /// The channel matching the channelId. To observe updates to the channel, set your class as a delegate of this controller and call `startUpdating`.
+    public var messages: [MessageModel<ExtraData>] {
+        guard state == .active else {
+            log.warning("Accessing `messages` before calling `startUpdating()` always results in an empty array.")
+            return []
         }
         
-        return aggregator
+        return messagesObserver.items
+    }
+    
+    /// The worker used to fetch the remote data and communicate with servers.
+    private lazy var worker: ChannelUpdater<ExtraData> = self.environment.channelUpdaterBuilder(client.databaseContainer,
+                                                                                                client.webSocketClient,
+                                                                                                client.apiClient)
+    
+    /// A type-erased delegate.
+    private(set) var anyDelegate: AnyChannelControllerDelegate<ExtraData>?
+    
+    private(set) lazy var channelObserver: EntityDatabaseObserver<ChannelModel<ExtraData>, ChannelDTO> = {
+        let observer = EntityDatabaseObserver(context: self.client.databaseContainer.viewContext,
+                                              fetchRequest: ChannelDTO.fetchRequest(for: self.channelQuery.cid),
+                                              itemCreator: ChannelModel<ExtraData>.create)
+        observer.onChange = { [unowned self] change in
+            self.delegateCallback { $0?.channelController(self, didUpdateChannel: change) }
```

Comment:
`channel` needs to be assigned
```
self.channel = change
```

Reply:
Nope, the observer takes care of this automatically.